File: February 28, 2026.js
function addPunctuation(sentences) {

  let arr = sentences.split("");
  const regexUpper = /^[A-Z]$/;

  for (let i = 0; i < arr.length; i++) {
    if(arr[i - 1] === " " && regexUpper.test(arr[i])) {
      arr.splice(i - 1, 0, ".")
      i++;
    }
  }

  if(!sentences.endsWith(".")) {
    arr.push(".");
  }

  let newSentence = arr.join("");

  return newSentence;
}

File: February 28, 2026.py
import re

def add_punctuation(sentences):
    arr = list(sentences)
    regex_upper = re.compile(r'^[A-Z]$')

    i = 0
    while i < len(arr):
        if i > 0 and arr[i - 1] == " " and regex_upper.match(arr[i]):
            arr.insert(i - 1, ".")
            i += 1
        i += 1

    if not sentences.endswith("."):
        arr.append(".")

    return "".join(arr)